state_machine/applications/flight/lib/logs.py

```diff
@@ -1,48 +1,63 @@
-from pycubed import cubesat
-from state_machine import state_machine
 import struct
 try:
     from ulab.numpy import array
 except ImportError:
     from numpy import array
+from pycubed import cubesat
+from state_machine import state_machine
 
 
-beacon_format = 'b' + 'f' * 11  # 1 char + 11 floats
+beacon_format = 3 * 'B' + 'H' + 'f' * 11  # 3 uint8 + 1 uint16 + 11 float32 = 49 bytes
 
-def beacon_packet(task):
-    """Creates a beacon packet containing the: CPU temp, IMU temp, gyro, acceleration, magnetic, and state byte.
-    The state byte is the index of the current state in the alphabetically ordered state list.
-    This data is packed into a c struct using `struct.pack`.
+def beacon_packet():
+    """Creates a beacon packet containing the: state index byte, f_contact and f_burn flags,
+    state_error_count, boot count, battery voltage,
+    CPU temperature, IMU temperature, gyro reading, mag reading,
+    radio signal strength (RSSI), radio frequency error (FEI).
 
-    If no IMU is attached it returns a packet of 0s.
+    This data is packed into a c struct using `struct.pack`.
     """
-    if not cubesat.imu:
-        task.debug('IMU not initialized')
-        return bytes([0, 0, 0, 0, 0])
-
+    state_byte = state_machine.states.index(state_machine.state)
+    flags = ((cubesat.f_contact << 1) | (cubesat.f_burn)) & 0xFF
```

Comment:
why not & 1

Reply:
Not sure what you mean by this? Do you mean to replace & 0xFF with & 0x01?

Reply:
This is setting both f_contact and f_burn as different bits in a single byte. The &0xff is really not necessary at all. It's mostly to guarantee that the result is a uint8. Using 0x01 doesn't make sense as it would zero f_contact. We could use 0x03.

Reply:
Oh I messed up parenthesis in my head